### ISSUE TITLE
Broken module error due to invalid LLVM integer arithmetic

### ISSUE THREAD

**Issue by s-ludwig** [ldc-developers/ldc#2537]
Running `dub test` on [`std_data_json`](https://github.com/s-ludwig/std_data_json) yields the following error:
```
Integer arithmetic operators only work with integral types!
  %6 = add i8* %4, %5, !dbg !5446
LLVM ERROR: Broken function found, compilation aborted!
```

Build logs: https://travis-ci.org/s-ludwig/std_data_json/builds/333671316

**Comment by kinke** [comment]
Thx for the report.

Failing command line:
```
ldc2 -c -unittest -Istd_data_json-master\source
    -I...\dub\packages\taggedalgebraic-0.10.9\taggedalgebraic\source
    std_data_json-master\source\stdx\data\json\value.d
```

Examplary offending IR with pointer arithmetic:
```LL
define weak_odr i8* @_D15taggedalgebraic__T12performOpRawTS4stdx4data4json5value9JSONValue12PayloadUnionVEQDf6OpKindi1VAyaa1_2bTnTQcZQDmFNaNbNiNfKQsQuZQx(i8* %_param_1_arg, i8** dereferenceable(8) %value) #0 comdat {
  %_param_1 = alloca i8*, align 8                 ; [#uses = 2, size/byte = 8]
  store i8* %_param_1_arg, i8** %_param_1
  %1 = load i8*, i8** %_param_1                   ; [#uses = 1]
  %2 = load i8*, i8** %value                      ; [#uses = 1]
  %3 = add i8* %1, %2                             ; [#uses = 1]
  ret i8* %3
}
```

**Comment by kinke** [comment]
The issue seems to boil down to
```D
void main()
{
    typeof(null) a, b;
    auto x = a + b;
}
```

apparently being valid code, but LDC trying to emit an addition of the two dummy pointers.

**Comment by kinke** [comment]
Okay, DMD swallows this too %)
```D
void main()
{
    typeof(null) a;
    int[string] dict;
    auto y = dict + a;
}
```

**Comment by kinke** [comment]
Fixed by #2540.
